Sources/Articles/Documentation.docc/2024/beginners-guide-to-protocol-buffers-and-grpc-with-swift/beginners-guide-to-protocol-buffers-and-grpc-with-swift.md

```diff
@@ -0,0 +1,382 @@
+    
+    var todos: [Todos_Todo]
+    
+    init(
+        todos: [Todos_Todo] = []
+    ) {
+        self.todos = todos
+    }
+    
+    func fetchTodos(
+        request: Todos_Empty,
+        context: GRPC.GRPCAsyncServerCallContext
+    ) async throws -> Todos_TodoList {
+        var result = Todos_TodoList()
+        result.todos += todos
+        return result
+    }
+    
+    func createTodo(
+        request: Todos_Todo,
+        context: GRPC.GRPCAsyncServerCallContext
+    ) async throws -> Todos_Todo {
+        todos.append(request)
+        return request
+    }
+    
+    func deleteTodo(
+        request: Todos_TodoID,
+        context: GRPC.GRPCAsyncServerCallContext
+    ) async throws -> Todos_Empty {
+        guard
+            let todo = todos.first(where: { $0.todoID == request.todoID })
+        else {
+            throw GRPCStatus.processingError
+        }
+        todos = todos.filter { $0.todoID != todo.todoID }
+        return .init()
+    }
+    
+    func completeTodo(
+        request: Todos_TodoID,
+        context: GRPC.GRPCAsyncServerCallContext
+    ) async throws -> Todos_Todo {
+        guard
+            var todo = todos.first(where: { $0.todoID == request.todoID })
+        else {
+            throw GRPCStatus.processingError
+        }
+        todo.completed = true
+        todos = todos.filter { $0.todoID != request.todoID }
+        todos.append(todo)
+        return todo
+    }
+}
+```
+
+The snippet above utilizes several types from the gRPC library, such as `GRPCStatus`, and `GRPCAsyncServerCallContext`, which is passed as an argument to each call. Additionally, the functions use the Swift data types generated from the `todo_messages.proto` definition, allowing you to provide the required input and output data.
+
+The final step is to configure the gRPC server. This can be done by using a an `Entrypoint` struct as a main entry point. We'll utilize the Swift Argument Parser library to allow users to specify both the hostname and port when launching the application. 
+
+This setup is based on the v1 gRPC library, which operates with `EventLoopFutures`. However, v2 is already in development and is built on modern Swift concurrency features, utilizing the Service Lifecycle library. As a result, this process will become much simpler once v2 is released.
+
+Below is an example of how to configure the server using the current gRPC release (v1):
+
+```swift
+import ArgumentParser
+import Logging
+import GRPC
+import NIO
+import GRPCCore
+
+@main
+struct Entrypoint: ParsableCommand {
+    
+    @Option(name: .shortAndLong)
+    var hostname: String = "127.0.0.1"
+    
+    @Option(name: .shortAndLong)
+    var port: Int = 1234
+    
+    func run() throws {
+        
+        var logger = Logger(label: "grpc-example")
+        logger.logLevel = .debug
+        
+        let group: EventLoopGroup = .singletonMultiThreadedEventLoopGroup
+
+        // 1. 
+        let server = Server.insecure(group: group)
```

Comment:
Add a comment in the code about insecure as well

Reply:
This is not an issue anymore in v2.